apps/native/app/src/screens/document-detail/document-detail.tsx

```diff
@@ -197,29 +205,35 @@ export const DocumentDetailScreen: NavigationFunctionComponent<{
   })
 
   const Document = {
-    ...(docRes.data?.getDocument || {}),
-    ...doc.data,
+    ...(doc?.data || {}),
+    ...(docRes.data?.documentV2 || {}),
+    ...(docContent.data?.documentV2 || {}),
   }
 
   useEffect(() => {
     if (doc.missing) {
-      void getListDocuments({
-        variables: {
-          input: {
-            page: 1,
-            pageSize: 50,
-          },
-        },
-      })
+      // If we don't have the document in the cache, fetch it
+      void getDocument()
+    } else if (doc.data?.content === null) {
+      // Content is missing, fetch it
+      void getDocumentContent()
```

Comment:
This looks like it has waterfall'y performance.

Maybe we should just combine the two in one lazy query which includes the document content and run it `if (doc.missing || doc.data?.content === null)`.

Reply:
Good point. Should we then fetch all of the data at all times? Because in most cases when we come from the inbox screen we have all the info we need except for `content.type` and `content.value`, so I was trying to prevent fetching all the other info when we already have it. I think in most cases we already have all the other data, the other call is just a 'belti and axlabönd' kind of thing 😄